lib/matplotlib/axis.py

```diff
@@ -711,6 +711,17 @@ def _set_scale(self, value, **kwargs):
     def limit_range_for_scale(self, vmin, vmax):
         return self._scale.limit_range_for_scale(vmin, vmax, self.get_minpos())
 
+    @property
+    def unit_data(self):
+        """Holds data that a ConversionInterface subclass relys on
+        to convert between labels and indexes
+        """
+        return self._unit_data
```

Comment:
You should create an empty `unit_data` in the `__init__` to avoid strange exception is someone calls the getter before the setter.